### PR TITLE
Submit: make url match reddit.com, some refactoring

### PR DESCRIPTION
**Pull request by curioussavage**
:eyeglasses: @ajacksified 

this was on my message-compose branch. Might as well get it out

now follows the same urls as .com, accepts query parameters to pre populate fields.

### CODE REVIEW

**Comment by ajacksified** [comment]
The link in the header still sends you to `/submit?subreddit=name` :nail_care:

**Comment by curioussavage** [comment]
:haircut: 

also fixed an issue with the setTitle function that was made visible by my sub-about branch

**Comment by curioussavage** [comment]
with this last commit, now depends on https://github.com/reddit/snoode/pull/50/commits.

File: assets/js/client.es6.js [hunk 1]
```diff
@@ -114,9 +114,10 @@ function modifyContext (ctx) {
   return ctx;
 }
 
-var $title = document.getElementsByTagName('title')[0];
+
 
 function setTitle(props={}) {
+  let $title = document.getElementsByTagName('title')[0];
```
Comment (curioussavage) [inline assets/js/client.es6.js]: was being called in a context where $title wasn't available

File: src/views/pages/submit.jsx [hunk 2]
```diff
@@ -150,18 +150,19 @@ class SubmitPage extends BasePage {
...
-      var message = err[1];
+    if (Array.isArray(err.errors)) {
+      let currentError = err.errors[0]
+      let type = currentError[0];
+      let message = currentError[1];
 
       if (type === 'BAD_CAPTCHA') {
         this.setState({
-          captchaIden: '',
+          captchaIden: err.captcha,
```
Comment (curioussavage) [inline src/views/pages/submit.jsx]: this saves a request for a new captcha identifier

**Comment by ajacksified** [comment]
:fish:

**Pull request merged**